### ISSUE TITLE
Pointcloud self filtering not working, seems to be due to parallelization

### ISSUE THREAD

**Issue by corot** [moveit/moveit_ros#511]
This a quite exotic problem that only happens to me on indigo sources. Deb packages and hydro source work fine.... what is puzzling cause there're no significant differences between indigo and hydro on the affected code!
So the problem is that [here](https://github.com/ros-planning/moveit_ros/blob/indigo-devel/perception/point_containment_filter/src/shape_mask.cpp#L151) the iterators over the pointcloud are not changing: point pt remains the same along the loop. I suppose it must be something related with the mysterious (for me) #pragma directive and the parallelization it enforces. However, removing the #pragma changes nothing!!!

So I just did the easy way: update the iterators in the old good way, and it works like a charm:

```
    sensor_msgs::PointCloud2ConstIterator<float> iter2_x(data_in, "x");
    sensor_msgs::PointCloud2ConstIterator<float> iter2_y(data_in, "y");
    sensor_msgs::PointCloud2ConstIterator<float> iter2_z(data_in, "z");

    // Cloud iterators are not incremented in the for loop, because of the pragma
#pragma omp parallel for schedule(dynamic)
    for (int i = 0 ; i < (int)np ; ++i)
    {
      Eigen::Vector3d pt = Eigen::Vector3d(*iter2_x, *iter2_y, *iter2_z);
      ++iter2_x; ++iter2_y; ++iter2_z;
      double d = pt.norm();
      ...
```

I'm pretty sure this must be something particular in my architecture. Maybe that I use an 8-cores processor, no idea. I have set PLATFORM_FAMILY and PLATFORM_NAME CMake variables as "amd" and "native" respectively, trying to optimize the code for my machine.
I don't wont to open any PR until I understand what's going on.

**Comment by mikeferguson** [comment]
@corot I think you are seeing something that we just fixed, see https://github.com/ros-planning/moveit_ros/pull/504, https://github.com/ros-planning/moveit_ros/issues/505 and the actual fix https://github.com/ros/common_msgs/pull/47.

TLDR: you need a source checkout of sensor_msgs until the changes propagate to debs.

**Comment by corot** [comment]
Exactly. I updated from shadow-fixed and now works like a charm. Thanks!
